.github/workflows/_build_container.yml

```diff
@@ -0,0 +1,65 @@
+name: Build NeMo Curator container
+on: 
+  workflow_call:
+    inputs:
+      ref: 
+        description: Git ref to checkout
+        default: ${{ github.sha }}
+        required: false
+        type: string
+
+defaults:
+  run:
+    shell: bash -x -e -u -o pipefail {0}
+
+jobs:
+  main:
+    runs-on: self-hosted
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: ${{ github.run_id }}
+          ref: ${{ inputs.sha }}
+
+      - name: Clean runner cache
+        run: | 
+          docker system prune --filter "until=24h" --force
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with: 
+          # We use `docker` driver as this speeds things up for 
+          # trivial (non-multi-stage) builds.
+          driver: docker
+
+      - name: Pull cache images
+        run: |
+          docker pull nemoci.azurecr.io/nemo_curator_container:${{ github.event.pull_request.number || 'buildcache' }} || true
+          docker pull nemoci.azurecr.io/nemo_curator_container:buildcache || true
+
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          file: Dockerfile
```

Comment:
This file needs to be created (on the root directory of the repository). Something simple like this will do the trick: 

```dockerfile
FROM python:3.10

WORKDIR /opt

ARG CURATOR_COMMIT
RUN <<"EOF" bash -exu
git clone https://github.com/NVIDIA/NeMo-Curator.git
cd NeMo-Curator
git checkout $CURATOR_COMMIT
pip install cython pytest setuptools
pip install --extra-index-url https://pypi.nvidia.com ".[cuda12x]"
EOF
```

It will build a self-contained image that we can later instantiate and run as a container